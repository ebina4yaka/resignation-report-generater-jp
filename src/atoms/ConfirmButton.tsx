import React from 'react'
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles'
import Button from '@material-ui/core/Button'
import clsx from 'clsx'

type Props = {
  text: string
  isDisable: boolean
  onClickFunction: () => void
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    margin: {
      margin: theme.spacing(2),
    },
    button: {
      width: '10ch',
    },
  })
)

export default function ConfirmButton(props: Props): React.ReactElement {
  const { text, isDisable, onClickFunction } = props
  const classes = useStyles()

  const renderButton = (): React.ReactElement => {
    if (isDisable) {
      return (
        <Button
          disabled
          className={clsx(classes.margin, classes.button)}
          variant="contained"
          color="primary"
          onClick={onClickFunction}
        >
          {text}
        </Button>
      )
    }
    return (
      <Button
        className={clsx(classes.margin, classes.button)}
        variant="contained"
        color="primary"
        onClick={onClickFunction}
      >
        {text}
      </Button>
    )
  }

  return renderButton()
}
