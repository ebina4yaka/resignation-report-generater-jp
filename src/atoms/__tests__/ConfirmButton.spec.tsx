import React from 'react'
import renderer from 'react-test-renderer'
import ConfirmButton from '../ConfirmButton'

test('Snapshot test for ConfirmButton Disable', () => {
  const tree = renderer
    .create(<ConfirmButton text="test" onClickFunction={() => {}} isDisable />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

test('Snapshot test for ConfirmButton Enable', () => {
  const tree = renderer
    .create(
      <ConfirmButton text="test" onClickFunction={() => {}} isDisable={false} />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
