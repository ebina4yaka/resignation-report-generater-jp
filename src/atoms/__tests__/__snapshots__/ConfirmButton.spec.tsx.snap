// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for ConfirmButton Disable 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-margin-1 makeStyles-button-2 MuiButton-containedPrimary Mui-disabled Mui-disabled"
  disabled={true}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={-1}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    test
  </span>
</button>
`;

exports[`Snapshot test for ConfirmButton Enable 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-margin-1 makeStyles-button-2 MuiButton-containedPrimary"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    test
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;
